### PR TITLE
Homework 5

### PR DESCRIPTION
**Pull request by korejur**
nisam baš uspjela ovaj dio s testom mutate funkcije kod gumba za brisanje. nije mi implementiran kao što je bilo u predavanju možda je do toga.

uglavnom htjela bih da to prokomentiramo na konzultacijama možda se može brzo riješiti samo ja ne znam sad kako. probala sam nešto s tim unutar ShowContainerReviewDeletion.test.tsx

### CODE REVIEW

File: tvshows/src/components/features/shows/ShowContainer/ShowContainer.tsx [hunk 1]
```diff
@@ -1,101 +1,67 @@
...
 
-  useEffect(() => {
-    if (showId) {
-      const loadedList = loadFromLocalStorage(Number(showId));
-      setReviewList(loadedList);
+  const addShowReview = async (review: IReview) => {
+    try {
+      await createReview(swrKeys.create_review, { arg: review });
+      console.log(reviewListResponse);
+      mutate(swrKeys.reviews(Number(showId)));
```
Comment (darrac) [inline tvshows/src/components/features/shows/ShowContainer/ShowContainer.tsx]: This `mutate` only changes the values in you internal SWR cache.
What you need to trigger a POST request to the server is `useSwrMutation` and its `trigger` function.

Also, you can then move this directly to the component that triggers, in this case, `ReviewForm`

File: tvshows/src/components/features/shows/ShowContainer/ShowContainer.tsx [hunk 2]
```diff
@@ -1,101 +1,67 @@
...
-  const loadFromLocalStorage = (showId: number) => {
-    const reviewListString = localStorage.getItem(`reviewlist-${showId}`);
-    if (!reviewListString) {
-      return mockReviewList;
-    }
+  const deleteShowReview = async (reviewToRemove: IReview) => {
     try {
-      return JSON.parse(reviewListString);
+      await deleteReview(swrKeys.review(Number(reviewToRemove.id)));
+      mutate(swrKeys.reviews(Number(showId)));
```
Comment (darrac) [inline tvshows/src/components/features/shows/ShowContainer/ShowContainer.tsx]: Same comment as for the add review. You can find an example of this in the lecture code.

**Review by darrac — CHANGES_REQUESTED** [review]
You need to update the way you use useSwr and useSwrMutation as follows:
1.  When you need to fetch data, use `useSwr`
2. When you need to POST, PATCH or DELETE something on the server, use `useSwrMutation`
3. After you change/add/delete some entities on the server with `useSWRMutation` tell all the components that render a list of those entities to update by calling `mutate`.

File: tvshows/src/components/features/review/ShowReviewSection/ShowReviewSection.tsx [hunk 3]
```diff
@@ -1,31 +1,22 @@
...
 }: IShowReviewSectionProps) => {
-
-  const handleAddReview = (review: IReview) => {
-    review.show_id = showId;
-    addShowReview(review);
-  }
   return (
     <>
-      <ReviewForm onAdd={handleAddReview} />
+      <ReviewForm showId = {showId} />
```
Comment (darrac) [inline tvshows/src/components/features/review/ShowReviewSection/ShowReviewSection.tsx]: Why not fetch the reviews in this component?

File: tvshows/src/components/features/shows/ShowContainer/ShowContainer.tsx
Comment (darrac) [inline tvshows/src/components/features/shows/ShowContainer/ShowContainer.tsx:1]: I don't think there is a need to calculate average rating anymore as you receive it from the API as part of the show resource.

File: tvshows/src/components/features/review/ShowReviewSection/ShowReviewSection.tsx (hunk 3 above)
Comment (korejur) [reply tvshows/src/components/features/review/ShowReviewSection/ShowReviewSection.tsx]: Ako raspodijelim fetchanje na ShowReviewSection i ShowDetailsCard onda više u ShowContaineru ne mogu provjeriti isLoading i error? Nakon te raspodjele, loading i error se provjeravaju u svakoj komponenti zasebno, ali onda mi se dvaput prikaže poruka Loading...
Što mi je činiti?

Comment (darrac) [reply tvshows/src/components/features/review/ShowReviewSection/ShowReviewSection.tsx]: To ti sada ne izgleda lijepo zato sto ti UI pokaze obicni `Loading` tekst. 
U stvarnom slucaju bi koristila https://v2.chakra-ui.com/docs/components/skeleton ili nesto slicno pa bi to OK izgledalo.
Nemoj se sada brinuti s tim sto se loading prikazuje 2 puta.

Comment (korejur) [reply tvshows/src/components/features/review/ShowReviewSection/ShowReviewSection.tsx]: Okej, pushala sam to. Nisam stigla sad u apartmanu primijenit to i u Homework 6 jer smo u nekoj žurbi bili, ali ako hoćeš, možeš ti to mergati. Jučer sam radila i na popravcima iz Homework 6 tako da bi to trebalo bit to kad se spoji s ovim najnovijim dijelom.

Comment (korejur) [reply tvshows/src/components/features/review/ShowReviewSection/ShowReviewSection.tsx]: Riješeno :)

**Review by darrac — APPROVED** [review]

**Pull request merged**